src/query-parser.ts

```diff
@@ -1,15 +1,39 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const dataTypes = {
+  double: (value) => Number(value),
+  int: (value) => Number(value),
+  float: (value) => Number(value),
+  long: (value) => Number(value),
+  boolean: (value) => Boolean(value)
+}
+
+const castType = (value, type) => {
+  if (!type) {
+    return value
+  }
+  const castFunction = dataTypes[type.split(' ')[0]]
+  return castFunction ? castFunction(value) : value
+}
+
 export class QueryParser {
+  private schemas: object
+
+  constructor(schemas) {
+    this.schemas = schemas
+  }
+
   parse(searchParams: Dictionary<any>): SearchQuery {
+
     if (searchParams.operator && searchParams.args) {
       return <SearchQuery>searchParams
     }
 
     const args: Array<any> = []
     for (const field in searchParams) {
-      const value = searchParams[field]
+      const typeLookUp = field == "type" ? null : this.schemas[field]
```

Comment:
type is not the only field from CanonicalEntity that's not described by the schema